### PR TITLE
Improve dynamic packet support

### PR DESCRIPTION
**Pull request by ryanmelt**
Notifies Logging and Decom that new packets have been added and they need to listen to more topics.
Python support not fully implemented.  Will write another ticket for that.

### CODE REVIEW

File: openc3/lib/openc3/packets/packet_config.rb [hunk 1]
```diff
@@ -315,30 +315,60 @@ def finish_packet
...
+        end
+
+        if affect_ids
+          hash = @tlm_id_value_hash[packet.target_name]
+          hash = {} unless hash
+          @tlm_id_value_hash[packet.target_name] = hash
+          update_id_value_hash(packet, hash)
+        end
+      end
+    end
```
Comment (jmthomas) [inline openc3/lib/openc3/packets/packet_config.rb:359]: Is there any other caching that needs updating or invalidating?

File: openc3/lib/openc3/system/system.rb [hunk 2]
```diff
@@ -72,6 +75,14 @@ def self.limits_set=(value)
       @@limits_set = value.to_s.intern
     end
 
+    def self.add_post_instance_callback(callback)
+      if @@instance
+        callback.call()
+      else
+        @@post_instance_callbacks << callback
+      end
+    end
```
Comment (jmthomas) [inline openc3/lib/openc3/system/system.rb:84]: What are the plans for this?

File: openc3/lib/openc3/microservices/microservice.rb [hunk 3]
```diff
@@ -210,5 +211,32 @@ def shutdown
...
+            @topics << new_topic unless @topics.include?(new_topic)
+          end
+        else
+          raise "Invalid topics given to microservice_cmd: #{topics}"
+        end
+        Topic.trim_topic(topic, msg_id)
+        return true
+      end
+      Topic.trim_topic(topic, msg_id)
+      return false
```
[the remark this answers is not in the document]
Comment (ryanmelt) [reply openc3/lib/openc3/microservices/microservice.rb:239]: Yes - Next command will be changing the Logger level.

File: openc3/lib/openc3/packets/packet_config.rb (hunk 1 above)
Comment (ryanmelt) [reply openc3/lib/openc3/packets/packet_config.rb:359]: I think I got everything...

File: openc3/lib/openc3/system/system.rb (hunk 2 above)
Comment (ryanmelt) [reply openc3/lib/openc3/system/system.rb:84]: I use it as a lifecycle hook to be able to define dynamic packets, after the regular system is fully created.

**Comment by ryanmelt** [comment]
Previously logging and decom worked on my first plugin, because I was dynamically adding fields to an existing packet, not creating new packets.   With new packets, they would not be logged or decommed without these changes.

**Pull request merged**